shared.go

```diff
@@ -199,6 +201,66 @@ type Header struct {
 	ExtraHeaders map[HeaderKey]interface{}
 }
 
+// Equal compares a header struct with another header struct
+func (h1 Header) Equal(h2 Header) bool {
```

Comment:
This adds `Header.Equal` as part of our public API. That might be something we should add anyhow, but I don't want to mix it in as part of a "remove a testing dependency" PR.

testify's version of `assert.Equal` delegates to [`reflect.DeepEqual`](https://pkg.go.dev/reflect#DeepEqual). I think that suffices for this struct since it recurses into slices and maps, and dereferences pointers. I'll push a diff to use DeepEqual directly and remove EqualFunc and this `Header.Equal`.